docs/standard/serialization/binaryformatter-migration-guide/wpf-applications.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "BinaryFormatter migration guide: WPF applications"
+description: "Learn about the effects of the deprecation and removal of BinaryFormatter from .NET on Windows Forms and how to migrate."
+ms.date: 8/05/2024
+no-loc: [BinaryFormatter, WPF]
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "WPF"
+---
+
+# WPF Migration Guide – Binary Formatter 
+
+ 
+## Binary Formatter Removal 
+
+In .NET 9.0, the entire `Binaryformatter` infrastructure will be removed from the product and moved as a separate Nuget  package. For more information about the risks `Binaryformatter` poses and our decision on its removal, see [BinaryFormatter migration guide](index.md). 
+
+ 
+## How Binary Formatter affects WPF 
+
+WPF uses `Binaryformatter` to serialize/deserialize the data for scenarios such clipboard, drag/drop and store data for Avalon Binding in Journal.  
+
+WPF has and is still currently using `Binaryformatter` code internally for all the above scenarios, however we have taken some measures to mitigate the usage of `Binaryformatter`.  
+
+We have added an implementation based on WinForms to handle common primitive types, including primitive arrays and lists. Usage of these types in the mentioned scenarios will continue to work without using `Binaryformatter`, similarly as before. 
+
+Primitive types include: 
+- `bool` 
+- `byte` 
+- `char` 
+- `decimal` 
+- `double` 
+- `int` 
+- `sbyte` 
+- `float` 
+- `TimeSpan` 
+- `DateTime` 
+- `uint` 
+- `string` 
+- `nint` 
+- `nuint` 
+- `long`
+- `ulong`
+- `short`
+- `ushort` 
+
+Additional types include: 
+- `PointF` 
+- `RectangleF`  
+  
+## Where Binary Formatter is used in WPF 
+
+### Clipboard  
```

Comment:
I had started adding clipboard section in `winforms-wpf-ole-guidance.md`. My understanding is that WinForms and WPF will hit BinaryFormatter in the same way. If you could review and let me know if it is accurate for WPF that would be great. Then, we can update this section to point to that doc instead.

Reply:
Thanks @lonitra for adding the common OLE scenarios for WPF and WinForms.
The changes are LGTM.
I have updated the section to point to the `winforms-wpf-ole-guidance.md`